src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -300,8 +307,13 @@ public void visitToken(DetailAST ast) {
             case TokenTypes.RECORD_DEF:
                 checkForRedundantModifier(ast, TokenTypes.FINAL, TokenTypes.LITERAL_STATIC);
                 break;
-            case TokenTypes.CLASS_DEF:
             case TokenTypes.VARIABLE_DEF:
+            case TokenTypes.PATTERN_VARIABLE_DEF:
+                if (jdkVersion >= JDK_21 && isUnnamedVariable(ast)) {
+                    checkForRedundantModifier(ast, TokenTypes.FINAL);
```

Comment:
Should/does this cover lambda parameters, catch parameters, and resources, which can be unnamed?
https://github.com/checkstyle/checkstyle/pull/15480
https://github.com/checkstyle/checkstyle/pull/15358
https://github.com/checkstyle/checkstyle/issues/15484

Reply:
Good point, This check already covers resources whether it is named or unnamed (BTW it isn't listed in the doc, in "The check validates" section maybe we should add it to this list in another PR)


but we need to support unnamed lambda parameters and  unnamed catch parameters

```
cat src/Test.java
import java.util.function.Function;

public class Test {
    
    void m() {
        try (final var a = lock();) { // violation

        } catch (final Exception e) {

        }

        try (final var _ = lock();) { // violation

        } catch (final Exception _) {

        }
    }

    void m2() {
        Function<Integer, Integer> f = (final Integer _) -> {
            return 5;
        };
    }

    AutoCloseable lock() {
        return null;
    }
}
D:\CS\test
java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:6:14: Redundant 'final' modifier. [RedundantModifier]
[ERROR] D:\CS\test\src\Test.java:12:14: Redundant 'final' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 2 errors.
```

Reply:
Done

Reply:
Please clarify what done means here. We added support for specific cases?

I ask because you mentioned lambda and catch but did not mention resources.

Reply:
done means that we add support for redundant final on unnamed lambda parameter and unnamed catch parameter, resource is already supported

Reply:
@mahfouz72 Can you updated regression with the new tokens added. I didn't see them listed at https://github.com/checkstyle/checkstyle/pull/15492#issuecomment-2307027002

Reply:
I also don't see new tokens at https://github.com/checkstyle/checkstyle/pull/15492/files#diff-9b10f3ecc777d61ee2044e7753ec6d50ea673d2d88ede4e44c87cc99f58ae733

Reply:
> Can you updated regression with the new tokens added

I generated another report but the workflow seems to be canceled I will regenerate now.

Reply:
> I also don't see new tokens at 

@rnveach done, because there was `(default)` the new default tokens were loaded even if we don't add them to the list in the config comment

Reply:
> @mahfouz72 Can you updated regression with the new tokens added. I didn't see them listed at [#15492 (comment)](https://github.com/checkstyle/checkstyle/pull/15492#issuecomment-2307027002)

https://github.com/checkstyle/checkstyle/pull/15492#issuecomment-2307569944